services/src/main/java/org/fao/geonet/api/processing/ValidateApi.java

```diff
@@ -183,27 +188,40 @@ public SimpleMetadataProcessingReport validateRecords(
             ServiceContext serviceContext = ApiUtils.createServiceContext(request);
 
             Set<String> records = ApiUtils.getUuidsParameterOrSelection(uuids, bucket, userSession);
+            report.setTotalRecords(records.size());
 
             for (String uuid : records) {
-                if (!metadataRepository.existsMetadataUuid(uuid)) {
-                    report.incrementNullRecords();
-                }
+                int loopConditionCount = 0;
                 for (AbstractMetadata record : metadataRepository.findAllByUuid(uuid)) {
-                    if (!accessMan.canEdit(serviceContext, String.valueOf(record.getId()))) {
-                        report.addNotEditableMetadataId(record.getId());
-                    } else {
-                        boolean isValid = validator.doValidate(record, serviceContext.getLanguage());
-                        if (isValid) {
-                            report.addMetadataInfos(record.getId(), "Is valid");
-                            new RecordValidationTriggeredEvent(record.getId(), ApiUtils.getUserSession(request.getSession()).getUserIdAsInt(), "1").publish(applicationContext);
+                    if (approved == null ||
+                        (approved == true && !(record instanceof MetadataDraft)) ||
+                        (approved == false && record instanceof MetadataDraft)) {
```

Comment:
This check is not correct, when created a new metadata the record is stored in the `Metadata` table, not in the `MetadataDraft` table. 

`metadataRepository.findAllByUuid(uuid)` should return 1 record if the metadata is approved or is draft, but never published and 2 records if the metadata is published and also has a draft copy. Not really optimal, but maybe can be used these conditions for the checks.

Reply:
Thinking of it some more I think I was trying to merger draft and approved into one and I need to use them separately.
I will modify the code so that approved=true only checks for approved records and approved=false checks for working copy.
Draft will also be a different check - which I think means not approved and not published.
I will add some checks to the metadatautil for checking is a metadata records isApproved, isPublished and isDraft.

I will push some changes for this shortly.